### DISCUSSION TITLE
Lazer consumes more CPU when the window is unfocused

### DISCUSSION THREAD

**Discussion by People-11** [ppy/osu#30192] [Q&A] [answered]
![Snipaste_2024-10-10_17-44-46_compressed](https://github.com/user-attachments/assets/9807f727-ea34-4c7f-a528-34a9dd4231ff)

As you can see in the GIF I uploaded, when I click on taskmgr (focus on it), the Lazer window is in background mode (still rendered but unfocused), but it instead consumes more CPU than the focused state. It's a bit counter-intuitive, and I'm not sure if it's normal or if it's Lazer's problem or caused by some other programs on my computer.

Logs:
[compressed-logs.zip](https://github.com/user-attachments/files/17323753/compressed-logs.zip)

**Comment by peppy** [comment]
You're sure it's lazer? Also have you ensured your clock speed isn't dropping into an idle state? If it is, the usage may look high while it's actually super low.

**Reply by People-11** [reply]
Yes, It's Lazer, the mouse cursor can prove it XD.
As you can see in the GIF, when I focus on taskmgr, the CPU usage increased to ~15% and clock speed boosted to 4GHz+, and when I focus back to the game it dropped to below 10% and 3.5GHz. Also when focus on taskmgr, there will have one core increase to full usage.

**Reply by peppy** [reply]
Yet you refuse to show the process tab showing that it's lazer using cpu? What does the mouse cursor prove?

I can't reproduce this FWIW.

**Reply by People-11** [reply]
I recorded a video using a 3rd party process monitor to show the process's cpu usage.
https://pixeldrain.com/u/SBQeo4t4 (updated with a more detailed CPU information)
As you can see when the music pause (game in background), osu!.exe do consumes a little bit more, and exactly I can hear my fan spins faster because one big core turns into full load. (12900H has big.little core design)
The reason why I don't use taskmgr to show the process list is some system component like xbox game bar will [show incorrect CPU usage](https://answers.microsoft.com/en-us/windows/forum/all/xbox-game-bar-not-showing-cpu-and-gpu-usage/bcad273a-3bea-4d5b-bd4f-a6fb4d0a9021) on big.little designs, and taskmgr's process list sometimes has this problem as well

https://pixeldrain.com/u/Jx19RjU1
This is a record shows taskmgr doesn't display correct CPU usage. But shows there are no other high usage programs running besides osu.

**Reply by peppy** [reply]
Thanks, this is helpful.

**Comment by Tenexxt** [comment]
Check out this awsome setting: 
![{5A578C0B-39EB-454A-8BAC-A5506DDA4762}](https://github.com/user-attachments/assets/af56e67e-effb-46c2-b91a-a6b6ec2afda2)
if you have this enabled then idk whats the problem, but if you don't, please enable it

**Reply by SchiavoAnto** [reply]
If the issue is legit, I don't see how your solution helps with it.

**Reply by Tenexxt** [reply]
The game is minimised a lot with this setting already off, but when turning it on it decreases even more (for me, minimised osu with this off takes around 8-5% CPU, with it on, it's less than 1%) so this could be useful if OP has this off.

The reason for the increased CPU% could be the fact Lazer is more GPU focused rather than CPU (if I remember correctly, if I'm incorrect about this then sorry) so maybe when the window is unfocused the GPU no longer is working on Lazer in the way it's supposed to, so the required power is taken from the CPU (this could be absolutely incorrect if Lazer is more CPU focused and I'm misremembering, but If I'm not maybe it's just some more settings to be tweaked, like this one)

**Reply by peppy** [reply]
Please take the handwavey bullshit somewhere else 😅 💦

**Comment by peppy** [comment]
Just as a heads-up, I am investigating this actively and can reproduce.

**Comment by peppy** [comment]
https://github.com/ppy/osu-framework/issues/3151 is likely tracking this same issue.

**Reply by People-11** [reply]
I'm still experiencing the ‘higher usage when window is not focused’ issue in the latest version, but the CPU usage when minimised in windowed mode is normal.

**Reply by peppy** [reply]
Nothing has changed yet. Please wait for some kind of resolution in the future.

**Comment by smoogipoo** [comment, marked as answer]
Will be fixed by https://github.com/ppy/osu-framework/pull/6451

**Reply by People-11** [reply]
Unfortunately the issue still exist on my environment in latest version. Should I do a clean install? Or @peppy do u still reproduce on latest version?

Update: Clean install doesn't fix the issue

**Reply by peppy** [reply]
You'll have to wait until the next release. It wasn't actually included in the current one due to issues.

**Comment by People-11** [comment]
The issue is fixed, thankyou!

**Reply by peppy** [reply]
Great to hear!
